types/chai/index.d.ts

```diff
@@ -791,7 +787,8 @@ declare namespace Chai {
          * @param value   Actual value.
          * @param message   Message to display on error.
          */
-        isBoolean<T>(value: T, message?: string): void;
+        isBoolean(value: object, message?: string): asserts value is never;
+        isBoolean(value: unknown, message?: string): asserts value is boolean;
```

Comment:
Here's a good example of a method that changed from using generics, to using a simple overload instead. `isBoolean` should never have been generic in the first place. 🤷 

With the overloads, we can make the case where a passed in value is `unknown` (or `any`) work correctly, as the later overload (the one that `asserts value is boolean`) will be used for those types for `value`.

I had to also add the `object` specific overload to handle object-like values correctly. Without that overload asserting that `value is never`, value would end up being asserted as a intersection between the original object type and `boolean`, which doesn't really make sense.